iml-agent/src/action_plugins/ntp/action_configure.rs

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::agent_error::ImlAgentError;
+use futures::{future, Future, Stream};
+use std::path::Path;
+
+static NTP_CONFIG_FILE: &'static str = "/etc/ntp.conf";
+static MARKER: &'static str = "# IML_EDIT";
+static REMOVE_MARKER: &'static str = "#REMOVE_MARKER#";
+static PREFIX: &'static str = "server";
+
+/// Gets a stream to the ntp config
+pub fn get_ntp_config_stream() -> impl Stream<Item = String, Error = ImlAgentError> {
+    iml_fs::stream_file_lines(Path::new(NTP_CONFIG_FILE)).from_err()
+}
+
+/// Writes the new config data to the config file
+pub fn update_and_write_new_config(
+    server: Option<String>,
+) -> impl Future<Item = (), Error = ImlAgentError> {
+    let s = get_ntp_config_stream();
+    configure_ntp(server, s)
+        .and_then(|updated_config| tokio::fs::write(NTP_CONFIG_FILE, updated_config).from_err())
+        .then(|_| Ok(()))
```

Comment:
Use `.map(drop)` here instead which won't swallow the error half of the future.

Reply:
This function changed to:
```
pub fn get_ntp_config_stream() -> impl Stream<Item = String, Error = ImlAgentError> {
    iml_fs::stream_file_lines(Path::new(NTP_CONFIG_FILE))
        .map_err(|_| panic!("Failed to get a stream to the ntp config file."))
}
```

Reply:
Should I still use a .map(drop)?

Reply:
Yes, using `then` will swallow any errors.